src/stores/gameList.js

```diff
@@ -52,14 +52,14 @@ export const useGameListStore = defineStore('gameList', {
     joinGame(data) {
       const updatedGame = this.openGames.find((game) => game.id === data.gameId);
       if (updatedGame) {
-        updatedGame.numPlayers = Math.min(2, updatedGame.numPlayers + 1);
+        updatedGame.players.push(data.newPlayer);
         updatedGame.status = data.newStatus;
       }
     },
     otherLeftGame(gameId) {
       const updatedGame = this.openGames.find((game) => game.id === gameId);
       if (updatedGame) {
-        updatedGame.numPlayers = Math.max(0, updatedGame.numPlayers - 1);
+        updatedGame.players.pop();
```

Comment:
I think this needs to be updated on the backend and here so that we can tell *which* player left and take the appropriate person out. It probably makes sense to emit their id in the socket event and use that to .filter() the array to exclude them.

The user flow to consider here is that the first player to join the game leaves ie

1. Player 1 joins
2. Player 2 joins
3. Player 1 leaves

In this situation we need to remove players[0] not players[1]. So we need to know who left specifically and then use that here to remove the right person

Reply:
This warrants a test case as well